### ISSUE TITLE
[Bug]: Automatic language selection from browser info does not work (and how to fix it)

### ISSUE THREAD

**Issue by pi-infected** [danny-avila/LibreChat#2023]
### What happened?

The UI is not able to pick the language from the browser info. 

In LibreChat/client/src/store/language.ts:

```
import { atom } from 'recoil';

const userLang = navigator.language || navigator.languages[0];

const lang = atom({
  key: 'lang',
  default: localStorage.getItem('lang') || userLang,
});

export default { lang };
```

The navigator.language returns 'fr' (and not 'fr-FR'). 

In LibreChat/client/src/localization/Translation.tsx:

the language map only allow 'fr-FR' format:

```
const languageMap: { [key: string]: unknown } = {
  'en-US': English,
  'ar-EG': Arabic,
  'zh-CN': Chinese,
  'de-DE': German,
  'es-ES': Spanish,
  'fr-FR': French,
  'it-IT': Italian,
  'pl-PL': Polish,
  'pt-BR': Portuguese,
  'ru-RU': Russian,
  'ja-JP': Japanese,
  'sv-SE': Swedish,
  'ko-KR': Korean,
  'zh-TC': TraditionalChinese,
  'vi-VN': Vietnamese,
  'tr-TR': Turkish,
  'nl-NL': Dutch,
  'id-ID': Indonesia,
  'he-HE': Hebrew,
}

```

To correct the problem, please add:

```
 'en': English,
  'ar': Arabic,
  'zh': Chinese,
  'de': German,
  'es': Spanish,
  'fr': French,
  'it': Italian,
  'pl': Polish,
  'pt': Portuguese,
  'ru': Russian,
  'ja': Japanese,
  'sv': Swedish,
  'ko': Korean,
  'vi': Vietnamese,
  'tr': Turkish,
  'nl': Dutch,
  'id': Indonesia,
  'he': Hebrew,
```

To the dictionary.

### Steps to Reproduce

1 - Have a french browser (or I guess any language other than english).
2 - Go to the LibreChat interface
3 - It is in english

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Berry-13** [comment]
After trying on both Chrome and Brave, I didn't experience any issues with the auto-select language. Can you answer this question? "What browsers are you seeing the problem on?"

**Comment by pi-infected** [comment]
I am on Firefox 123.0.1 (64 bits)

**Comment by Berry-13** [comment]
> I am on Firefox 123.0.1 (64 bits)

wow, it's quite strange that in Italian it's working but not with French

**Comment by pi-infected** [comment]
![Screenshot_2024-03-08-18-38-02_1920x1080](https://github.com/danny-avila/LibreChat/assets/95248466/33ccc8dc-ab7d-4009-b49a-99bcb6020af8)

Mmm... I think I understand. A while ago I added french language manually to my browser because my system was in english. I changed the order of strings so 'fr' comes before 'fr-FR'. I think it is the other way around by default, so when the javascript asks the language it receive 'fr' instead of 'fr-FR' which probably have not happened in your tests. 

Anyway, I guess the fix I proposed is simple and good for other weird cases too. I let you decide ;)
